tests/conftest.py

```diff
@@ -300,14 +310,64 @@ def job_registry() -> InMemoryJobRegistry:
 
 
 @pytest.fixture
-def backend_implementation(batch_job_output_root, job_registry) -> "GeoPySparkBackendImplementation":
+def _dynamic_overrides(idp_config_file) -> typing.Generator[dict, None, None]:
+    """
+    Some overrides are generated files and need to be in config. Those will require handling in here
+    """
+    overrides = {}
+    if idp_config_file is not None:
+        overrides["openeo_idp_details_file"] = idp_config_file
+    if moto_server_address is not None:
+        overrides["s3_region_proxy_endpoints"] = {
+            "eu-central-1": moto_server_address
+        }
+    yield overrides
+
+
+@pytest.fixture
+def config_overrides() -> dict:
+    # Default implementation does not have overrides
+    return {}
+
+
+@pytest.fixture
+def _config_overrides(config_overrides, _dynamic_overrides) -> dict:
+    # use config_overrides to set manual overrides for your test
+    return {
+        **_dynamic_overrides,
+        **config_overrides
+    }
+
+
+@pytest.fixture
+def set_config_overrides(_config_overrides) -> typing.Generator[GpsBackendConfig, None, None]:
+    openeo_driver_overrides = {}
+    for config_key, config_override_value in _config_overrides.items():
+        if hasattr(OpenEoBackendConfig, config_key):
+            openeo_driver_overrides[config_key] = config_override_value
+
+    geopyspark_driver_overrides = {}
+    for config_key, config_override_value in _config_overrides.items():
+        if hasattr(OpenEoBackendConfig, config_key):
+            openeo_driver_overrides[config_key] = config_override_value
+        elif hasattr(openeogeotrellis.config.GpsBackendConfig, config_key):
+            geopyspark_driver_overrides[config_key] = config_override_value
+        else:
+            raise ValueError(f"config value {config_key} cannot be attributed to a config")
+    with openeo_driver.testing.config_overrides(**openeo_driver_overrides):
+        with openeogeotrellis.testing.gps_config_overrides(**geopyspark_driver_overrides, **openeo_driver_overrides):
+            yield get_backend_config()
```

Comment:
This config override system is getting quite complex with lot of layers. I think we should find a way to simplify it

Reply:
I have pushed one more commit to https://github.com/Open-EO/openeo-geopyspark-driver/pull/1206 to remove 1 of the fixtures (inline it) and make it clear which fixtures are internal